lms/djangoapps/grades/grade_utils.py

```diff
@@ -16,3 +148,62 @@ def are_grades_frozen(course_key):
             now = timezone.now()
             return now > freeze_grade_date
     return False
+
+
+def set_score(usage_key, student_id, score, max_points, **defaults):
+    """
+    Set a score.
+    """
+    if not isinstance(usage_key, UsageKey):
+        usage_key = UsageKey.from_string(usage_key)
+    defaults['module_type'] = 'problem'
+    defaults['grade'] = score / max_points
+    defaults['max_grade'] = max_points
+    StudentModule.objects.update_or_create(
+        student_id=student_id,
+        course_id=usage_key.course_key,
+        module_state_key=usage_key,
+        defaults=defaults)
+
+
+def get_score(usage_key, user_id):
+    """
+    Return score for user_id and usage_key.
+    """
+    if not isinstance(usage_key, UsageKey):
+        usage_key = UsageKey.from_string(usage_key)
+    try:
+        score = StudentModule.objects.get(
+            course_id=usage_key.course_key,
+            module_state_key=usage_key,
+            student_id=user_id
+        )
+    except StudentModule.DoesNotExist:
+        return None
+    else:
+        return {
+            'grade': score.grade,
+            'score': score.grade * (score.max_grade or 1),
+            'max_grade': score.max_grade,
+            'created': score.created,
+            'modified': score.modified
+        }
+
+
+def get_scores(usage_key, user_ids=None):
+    """
+    Return dictionary of student_id: scores.
+    """
+    scores_qset = StudentModule.objects.filter(
+        course_id=usage_key.course_key,
+        module_state_key=usage_key,
+    )
+    if user_ids:
+        scores_qset = scores_qset.filter(student_id__in=user_ids)
+
+    return {row.student_id: {'grade': row.grade,
+                             'score': row.grade * (row.max_grade or 1),
+                             'max_grade': row.max_grade,
+                             'created': row.created,
+                             'modified': row.modified,
+                             'state': row.state} for row in scores_qset}
```

Comment:
Could you put the `for row in scores_qset` on a new line?  It gets lost pretty easily in the dictionary construction like this.